models/db_test.go

```diff
@@ -0,0 +1,223 @@
+package models_test
+
+import (
+	"database/sql"
+	"log"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/PhilLar/Images-back/mocks"
+	"github.com/PhilLar/Images-back/models"
+	"github.com/golang/mock/gomock"
+
+	// The "testify/suite" package is used to make the test suite
+	"github.com/stretchr/testify/suite"
+)
+
+type StoreSuite struct {
+	suite.Suite
+	/*
+		The suite is defined as a struct, with the store and db as its
+		attributes. Any variables that are to be shared between tests in a
+		suite should be stored as attributes of the suite instance
+	*/
+	store *models.Store
+	db    *sql.DB
+}
+
+func (s *StoreSuite) SetupSuite() {
+	/*
+		The database connection is opened in the setup, and
+		stored as an instance variable,
+		as is the higher level `store`, that wraps the `db`
+	*/
+	connString := "postgres://images:images_go@localhost/images_test?sslmode=disable"
+	db, err := sql.Open("postgres", connString)
+	if err != nil {
+		s.T().Fatal(err)
+	}
+	s.db = db
+	dbPsql, err := models.NewDB(connString, "file://../migrations")
+	if err != nil {
+		log.Printf(err.Error())
+	}
+
+	s.store = &models.Store{DB: dbPsql}
+	//s.store, err = models.NewDB(connString, "file://../migrations")
+	//if err != nil {
+	//	log.Printf(err.Error())
+	//}
+}
+
+func (s *StoreSuite) SetupTest() {
+	/*
+		We delete all entries from the table before each test runs, to ensure a
+		consistent state before our tests run. In more complex applications, this
+		is sometimes achieved in the form of migrations
+	*/
+	_, err := s.db.Query("DELETE FROM images")
+	if err != nil {
+		s.T().Fatal(err)
+	}
+}
+
+func (s *StoreSuite) TearDownSuite() {
+	// Close the connection after all tests in the suite
+	s.db.Close()
+}
+
+func TestStoreSuite(t *testing.T) {
+	s := new(StoreSuite)
+	suite.Run(t, s)
+}
+
+func (s *StoreSuite) TestInsertImage() {
+	// Create a bird through the store `CreateBird` method
+	s.T().Run("Correct insertion", func(t *testing.T) {
```

Comment:
Use `s.Run()` instead:
```suggestion
	s.Run("Correct insertion", func(t *testing.T) {
```